frontend/src/app/components/blocks-list/blocks-list.component.ts

```diff
@@ -52,6 +56,20 @@ export class BlocksList implements OnInit {
 
     if (!this.widget) {
       this.websocketService.want(['blocks']);
+      this.blocksCountInitialized$
+      .pipe(
+        filter(blocksCountInitialized => blocksCountInitialized),
+        take(1)
+      ).subscribe(() => {
```

Comment:
It's wrong to nest a subscription inside a subscription.
Just use `switchMap() => this.route.queryParams`